HHS/Head-Start-TTADP
# [TTAHUB-2391] Merging curated goals feature flag and logic

## Description of changep

> I think we could just use the most recent root cause. If they closed and opened a new one with a new root, then we should use that one. - Tammy

Changes:

- Add feature flag: `closed_goal_merge_override`
- Similarity result aggregator checks for the feature flag, and if you have it, then we don't exclude closed goals in the list of available goals to merge.
- On the FE, we now auto select the latest curated goal (as the final goal) when more than one curated goals are selected.
- The goal selected as the final goal is the one whose root causes will be used for the newly created goal, the other root causes are discarded.

## How to test

In the most recent prod dump, go to the RTR for recipient 1884, region 8. Without this change, you would see one group of similar goals. With this change you would see two groups of similar goals if you have the feature flag or are an admin.

Selecting the first group, you see one in progress FEI goal, and one closed FEI goal. Go through the merge process. Whichever goal you choose as the final goal, you will keep those root causes.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2391


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
